### DISCUSSION TITLE
Lesson 9: Failing to claim the lottery because the user is offline

### DISCUSSION THREAD

**Discussion by demhack012** [Cyfrin/foundry-full-course-cu#1378] [Q&A] [answered]
we know form lesson 4 fund me that when you click withdraw a meta mask popup will appear for you to pay the gas price of the transaction. My question is what if the lottery chose a winner but he was offline and failed to pay the gas price of the transaction what will happen?

**Comment by Niferu** [comment, marked as answer]
Hello @demhack012 

There is no such thing as being offline in blockchain, at least not in this particular example. You have to understand who is calling what, in this case our `fulfillRandomWords` function is sending money to winner after picking one. This is the function that Chainlink VRF node calls. If we have automated our contract with Chainlink Keepers, we have to send some LINK tokens, so `fulfillRandomWords` can be called by Chainlink Automation contract. The LINK tokens held in your account are used to pay for the gas fees associated with calling functions like CheckUpkeep and PerformUpkeep. This way, the maintenance and automation of your contract are funded by the LINK balance you've allocated for the Chainlink Keeper service, any other fees are deducted from it too. Otherwise if contract is not automated you have to trigger transaction manually and that is where you pay gas cost directly. To sum up winner is not paying anything as he did not trigger transaction.

**Reply by demhack012** [reply]
I got it now thanks so the gas fees are paid by the address that calls the function on a smart contract I misunderstood this at the beginning thanks.

**Reply by Niferu** [reply]
@demhack012 

Yes, gas cost is always paid by primal/original caller of function. Case with Chainlink Keepers is a bit different as I explained above :) I'm glad I could help wish you best luck with learning cheers!
